### PR TITLE
feat(middleware): added restrictToOwnerOrAdmin middleware to the authController.js file

### PR DESCRIPTION
**Pull request by PieroCarimi**
Added the restrictToOwnerOrAdmin middleware to the authController.js file, and added that middleware to the GET, PATCH, and DELETE methods of the /activities/:id route in the ActivityRoute.js file and to the GET, PATCH, and DELETE methods of the /users/:id route.

### CODE REVIEW

**Review by Aierbote — APPROVED** [review]
That might be the longest function name till now XD. That aside, all right !

**Review by RiccardoGrimaudo — APPROVED** [review]

**Review by alessandrocolla — APPROVED** [review]
All good

File: controllers/authController.js [hunk 1]
```diff
@@ -176,3 +176,18 @@ exports.loginAuth = catchAsync(async (req, res, next) => {
 
   next();
 });
+
+exports.restrictToOwnerOrAdmin = (Model) =>
+  catchAsync(async (req, res, next) => {
+    const document = await Model.findById(req.params.id);
+
+    if (!document) {
+      return next(new AppError("No activity found with that ID", 404));
```
Comment (Aierbote) [inline controllers/authController.js]: There still is `activity` in this AppError

**Review by Aierbote — COMMENTED** [review]
There is just one small change to be made in `authController.js`

**Pull request merged**